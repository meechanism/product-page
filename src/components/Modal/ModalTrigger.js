import React from 'react';
import Button from '../Button';

export const onClick = (showModal) => (e) => {
  e.preventDefault();
  showModal();
};

const ModalTrigger = ({ buttonRef, showModal, triggerText }) => (
  <Button secondary ref={buttonRef} onClick={onClick(showModal)}>
    {triggerText}
  </Button>
);

export default ModalTrigger;
