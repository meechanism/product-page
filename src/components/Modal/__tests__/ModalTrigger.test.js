import React from 'react';
import ModalTrigger, { onClick } from '../ModalTrigger';

describe('ModalTrigger component', () => {
  it('should render', () => {
    const output = shallow(<ModalTrigger />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should trigger callback', () => {
    const mockFn = jest.fn();
    const e = { preventDefault: jest.fn() };
    onClick(mockFn)(e);
    expect(mockFn).toHaveBeenCalled();
  });
});
