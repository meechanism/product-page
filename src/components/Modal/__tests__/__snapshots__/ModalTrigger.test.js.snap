// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalTrigger component should render 1`] = `
<styled.button
  onClick={[Function]}
  secondary={true}
/>
`;
